src/axom/primal/geometry/BezierCurve.hpp

```diff
@@ -174,11 +264,18 @@ class BezierCurve
   /// Retrieves the control point at index \a idx
   const PointType& operator[](int idx) const { return m_controlPoints[idx]; }
 
+  /// Get specific weight
+  const T& getWeight(int idx) const { return m_weights[idx]; }
+
+  /// Change specific weight
+  void setWeight(int idx, T weight) { m_weights[idx] = weight; };
```

Comment:
Perhaps, e.g. by adding an `SLIC_ASSERT(this->isRational())` and a `\pre` precondition clause to the doxygen comment.

Reply:
These `SLIC_ASSERT` statements don't get run on a release build, correct? Would a proper `if( !isRational() )` statement for each call just be too expensive at runtime?

Reply:
The idea is that we guard against this in debug builds but don't pay the performance hit during release builds.

I.e. calling this function on a non-rational curve is a developer error, and we assume that at runtime people know what they're doing, but we want to help them debug if there's a problem.

Reply:
Another way to think about it is that it might be worse to silently not set the weight for non-rational curves than to allow the code to segfault, since the latter might cause the code to crash in a later, unrelated place.